x-template.js

```diff
@@ -875,11 +868,22 @@ class TemplateEngine {
   }
 
   static #createRawResult(language, strings, values) {
-    return { [TemplateEngine.#LANGUAGE]: language, strings, values };
+    const analysis = TemplateEngine.#setIfMissing(TemplateEngine.#stringsToAnalysis, strings, () => ({}));
+    if (!analysis.done) {
+      const fragment = TemplateEngine.#createFragment(language, strings);
+      const lookups = TemplateEngine.#findLookups(fragment);
+      analysis.language = language;
+      analysis.fragment = fragment;
+      analysis.lookups = lookups;
+      analysis.done = true;
+    }
+    // This is a leaking implementation detail, but fixing the leak comes at
+    //  a non-negligible performance cost.
+    return { [TemplateEngine.#ANALYSIS]: analysis, strings, values };
```

Comment:
Technically, you could `.getOwnPropertySymbols()` to find the non-enumerable “hidden” symbols like the one we use here to gain access to our internal `analysis`. That feels like a bit of a stretch to be concerned about though… practically-speaking.